packages/synchro-charts/cypress/integration/dial/dial.spec.ts

```diff
@@ -0,0 +1,397 @@
+    },
+  ],
+};
+
+it('renders latest value', () => {
+  const LATEST_VALUE1 = 100;
+  const LATEST_VALUE2 = 2238;
+  const DATA = round((LATEST_VALUE2 / (Y_MAX - Y_MIN)) * 100);
+  visitDynamicWidget(cy, {
+    componentTag: 'sc-dial',
+    size: DIAL_SIZE,
+    dataStream: {
+      ...DATASTREAM,
+      data: [
+        { x: new Date(1999, 0, 0).getTime(), y: LATEST_VALUE1 },
+        { x: new Date(2000, 0, 0).getTime(), y: LATEST_VALUE2 },
+      ],
+    },
+    ...VIEWPORT,
+  });
+  cy.wait(1000);
+  cy.get('.sc-dialbase-container')
+    .invoke('text')
+    .should('contain', `${DATA}%`);
+
+  cy.get(VALUE_LOADING).should('not.exist');
+  cy.get(VALUE_ERROR).should('not.exist');
+
+  cy.matchImageSnapshotOnCI();
+});
+
+it('renders value under percentile when detailedName is existed', () => {
+  const LATEST_VALUE1 = 100;
+  const LATEST_VALUE2 = 2238;
+  const DATA = round((LATEST_VALUE2 / (Y_MAX - Y_MIN)) * 100);
+  visitDynamicWidget(cy, {
+    componentTag: 'sc-dial',
+    size: DIAL_SIZE,
+    dataStream: {
+      ...DATASTREAM,
+      data: [
+        { x: new Date(1999, 0, 0).getTime(), y: LATEST_VALUE1 },
+        { x: new Date(2000, 0, 0).getTime(), y: LATEST_VALUE2 },
+      ],
+      detailedName: 'Medium',
+    },
+    ...VIEWPORT,
+  });
+  cy.wait(1000);
+  cy.get('.sc-dialbase-container')
+    .invoke('text')
+    .should('contain', `${DATA}%`);
+
+  cy.get(VALUE_LOADING).should('not.exist');
+  cy.get(VALUE_ERROR).should('not.exist');
+
+  cy.matchImageSnapshotOnCI();
+});
+
+it('renders value under percentile when significantDigits is 2', () => {
+  const significantDigits = 2;
+  const LATEST_VALUE = 2238;
+  const DATA = ((LATEST_VALUE / (Y_MAX - Y_MIN)) * 100).toPrecision(significantDigits);
+  visitDynamicWidget(cy, {
+    componentTag: 'sc-dial',
+    size: DIAL_SIZE,
+    significantDigits,
+    dataStream: {
+      ...DATASTREAM,
+      data: [{ x: new Date(2000, 0, 0).getTime(), y: LATEST_VALUE }],
+    },
+    ...VIEWPORT,
+  });
+  cy.wait(1000);
+  cy.get('.sc-dialbase-container')
+    .invoke('text')
+    .should('contain', `${DATA}%`);
+
+  cy.get(VALUE_LOADING).should('not.exist');
+  cy.get(VALUE_ERROR).should('not.exist');
+
+  cy.matchImageSnapshotOnCI();
+});
+
+it('renders error message when value is string', () => {
+  const LATEST_VALUE = 'ABC';
+  visitDynamicWidget(cy, {
+    componentTag: 'sc-dial',
+    size: DIAL_SIZE,
+    dataStream: {
+      ...DATASTREAM,
+      unit: 'rpm',
+      data: [{ x: new Date(1999, 0, 0).getTime(), y: LATEST_VALUE }],
+      dataType: DataType.STRING,
+    },
+    ...VIEWPORT,
+  });
+
+  cy.wait(1000);
+  cy.contains('.sc-dialbase-container', NO_VALUE_PRESENT).should('be.visible');
+  cy.get(VALUE_ERROR)
+    .invoke('text')
+    .should('contain', 'Invalid value');
+
+  cy.get(VALUE_LOADING).should('not.exist');
+
+  cy.matchImageSnapshotOnCI();
+});
+
+it('renders error message when value is not included in the range of `yMin` and `yMax`', () => {
```

Comment:
values outside of the range defined by `[yMin, yMax]` should not be represented by an error state. for example, a value equal to 2 * yMax should be represented as `200%`.

Reply:
For example, if yMin = 0, yMax = 100, if the value is equal to -50, then we'd display `-50%`

Reply:
If value is 0% or less, show empty ring, if value is 100% or greater, full ring

Reply:
Done. Add related code and test.